snmp/datadog_checks/snmp/snmp.py

```diff
@@ -112,93 +85,180 @@ def create_snmp_engine(self, mibs_path):
+                if 'symbol' in metric:
+                    to_query = metric['symbol']
+                    try:
+                        table_oids.append(hlapi.ObjectType(hlapi.ObjectIdentity(metric['MIB'], to_query)))
+                    except Exception as e:
+                        warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
+                elif 'symbols' not in metric:
+                    raise ConfigurationError('When specifying a table, you must specify a list of symbols')
+                else:
+                    for symbol in metric['symbols']:
+                        try:
+                            table_oids.append(hlapi.ObjectType(hlapi.ObjectIdentity(metric['MIB'], symbol)))
+                        except Exception as e:
+                            warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
+                    if 'metric_tags' in metric:
+                        for metric_tag in metric['metric_tags']:
+                            if not ('tag' in metric_tag and ('index' in metric_tag or 'column' in metric_tag)):
+                                raise ConfigurationError(
+                                    'When specifying metric tags, you must specify a tag, and an index or column'
+                                )
+                            if 'column' in metric_tag:
+                                # In case it's a column, we need to query it as well
+                                try:
+                                    table_oids.append(
+                                        hlapi.ObjectType(hlapi.ObjectIdentity(metric['MIB'], metric_tag.get('column')))
+                                    )
+                                except Exception as e:
+                                    warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
+
+            elif 'OID' in metric:
+                raw_oids.append(hlapi.ObjectType(hlapi.ObjectIdentity(metric['OID'])))
+            else:
+                raise ConfigurationError('Unsupported metric in config file: {}'.format(metric))
+
+        return table_oids, raw_oids, mibs_to_load
+
+
+class SnmpCheck(AgentCheck):
+
+    SOURCE_TYPE_NAME = 'system'
+    # pysnmp default values
+    SC_STATUS = 'snmp.can_check'
+
+    def __init__(self, name, init_config, instances):
+        # Set OID batch size
+        self.oid_batch_size = int(init_config.get('oid_batch_size', DEFAULT_OID_BATCH_SIZE))
+
+        # Load Custom MIB directory
+        self.mibs_path = None
+        self.ignore_nonincreasing_oid = False
+        if init_config is not None:
+            self.mibs_path = init_config.get('mibs_folder')
+            self.ignore_nonincreasing_oid = is_affirmative(init_config.get('ignore_nonincreasing_oid', False))
+
+        self._conf = {}
+
+        for instance in instances:
+            self._load_conf(instance)
```

Comment:
WDYTA storing state here instead, like we're doing with other Agent 6+ only checks (all checks on master now)? See disk, ibm_db2, tls, etc.